dash/App/Chains/Chain/Components/index.js

```diff
@@ -0,0 +1,213 @@
+import React from 'react'
+
+import link from '../../../../../resources/link'
+import svg from '../../../../../resources/svg'
+import RingIcon from '../../../../../resources/Components/RingIcon'
+
+const networkDefaults = {
+  type: 'ethereum',
+  id: 'Chain ID',
+  name: 'Chain Name',
+  explorer: 'Block Explorer',
+  symbol: 'Native Symbol',
+  isTestnet: false,
+  color: 'accent3'
+}
+
+export const SubmitChainButton = ({ text, enabled, textColor, onClick }) => {
+  return (
+    <div
+      role='button'
+      className={enabled ? 'addTokenSubmit addTokenSubmitEnabled' : 'addTokenSubmit'} 
+      style={{ color: textColor }}
+      onClick={(e) => {
+        // Left Click
+        if (e.button === 0) onClick()
+      }}
+    >
+      <span>{text}</span>
+    </div>
+  )
+}
+
+export const ChainHeader = ({ type, id, primaryColor, icon, svgName, name, on, showExpand, showToggle }) => {
+  return (
+    <div className='signerTop'>
+      <div className='signerDetails'>
+        <div className='signerIcon'>
+          <RingIcon 
+            color={`var(--${primaryColor})`}
+            img={icon}
+            svgName={svgName}
+          />
+        </div>
+        {/* <div className='signerType' style={this.props.inSetup ? {top: '21px'} : {top: '24px'}}>{this.props.model}</div> */}
+        <div className='signerName'>
+          {name}
+        </div>
+      </div>
+      <div className='signerMenuItems'>
+        {showExpand ? (
+          <div className='signerExpand' onClick={() => {
+            const chain = { id, type }
+            link.send('tray:action', 'navDash', { view: 'chains', data: { selectedChain: chain } })
+          }}>
+            {svg.bars(14)}
+          </div>
+        ) : null}
+        {showToggle ? <div 
+          className={on ? 'signerPermissionToggle signerPermissionToggleOn' : 'signerPermissionToggle'} 
+          onClick={id !== 1 ? () => link.send('tray:action', 'activateNetwork', type, id, !on) : null}
+        >
+          {id === 1 ? (
+            <div className='signerPermissionToggleSwitchLocked'>
+              {svg.lock(10)}
+              <div className='signerPermissionToggleSwitch' />
+            </div>
+          ) : (
+            <div className='signerPermissionToggleSwitch' />
+          )}
+        </div> : null}
+      </div>
+    </div>
+  )
+}
+
+export const EditChainColor = ({currentColor, onChange}) => {
+  return (
+    <div className='chainRow'>
+      <div className='chainInputLabel'>Chain Color</div>
+      <div className='chainColorSwatches'>
+        {[ 'accent1', 'accent2', 'accent3', 'accent4', 'accent5', 'accent6', 'accent7', 'accent8'].map(color => {
```

Comment:
might be cleaner to do this above instead of inline. extract the different accent values into a constant and then it can be something like:
```
const colorSwatches = accents.map( ... )

return (
  <div>{colorSwatches}</div>
)
```